### ISSUE TITLE
You should be able to set custom model names from getModelForClass overwriteOptions, not just @modelOptions decorator

### ISSUE THREAD

**Issue by benedict-jackson** [typegoose/typegoose#482]
Currently the getName function used to determine the model's name only looks at the `@modelOptions` keys, rather than the merged modeloptions and overwrite options. I've included the relevant bit of code from the getName function.

```ts
const ctor: any = getRightTarget(cl);
const options: IModelOptions = Reflect.getMetadata(DecoratorKeys.ModelOptions, ctor) ?? {};
const baseName: string = ctor.name;
const customName = options.options?.customName;
```

This means currently any custom naming properties passed into the `getModelForClass` function are ignored. It would be nice to be able to create multiple models with different names/collections off the same class without having to do lots of extending and adding `modelOptions`.

**Comment by benedict-jackson** [comment]
Seems like this isn't quite as straightforward as it first looks, because getName is used in all the class decorators, I'm not sure it'd actually be possible to defer all the name generation until the model is actually ready to be compiled.

It might be useful to add something to the docs to state that setting a new name is only possible in @modelOptions and taking customName + automaticName out of the type interface for the getModelForClass options.

**Comment by hasezoey** [comment]
Possible duplicate of #401

**Comment by benedict-jackson** [comment]
I think my use case is slightly different (wanting two collections in the same database off one class) whereas I assumed #401 was closer to #419 (same class two databases). Seems like it might be the same thing causing the issue though, the name is necessary to manage how all the other decorators are applied and how the model is registered with mongoose.

**Comment by smolinari** [comment]
> wanting two collections in the same database off one class

Couldn't you just extend from the one class and give it a different collection name? Then generate that other model too?

Scott

**Comment by benedict-jackson** [comment]
@smolinari That's what I've ended up doing, but it's much more long winded.

**Comment by smolinari** [comment]
Well, it shows intent though. Remember, code that you write now isn't just for a machine. It's more for other humans to read, including your own future self, which means you shouldn't be lazy to yourself either. LOL! 😁 

Scott

**Comment by hasezoey** [comment]
now that i re-read this issue, it seems to be fixed by #502
